### PR TITLE
Turn on serialization by default for JsonView/XmlView.

### PR DESCRIPTION
**Pull request by ADmad**
Refs cakephp/cakephp#6502

### CODE REVIEW

File: src/Controller/AppController.php [hunk 1]
```diff
@@ -39,4 +40,17 @@ public function initialize()
...
+     * Before render callback.
+     *
+     * @param \Cake\Event\Event $event The beforeRender event.
+     * @return void
+     */
+    public function beforeRender(Event $event)
+    {
+        if ($this->viewClass === 'JsonView') {
+            $this->set('_serialize', true);
+        }
```
Comment (markstory) [inline src/Controller/AppController.php]: What about xmlview?

Comment (josegonzalez) [inline src/Controller/AppController.php]: Also, what if the fully-qualified namespace is used?

Comment (ADmad) [inline src/Controller/AppController.php]: > What about xmlview?

The check should be updated when same feature is implemented for XmlView too :)

> Also, what if the fully-qualified namespace is used?

RequestHandler sets the viewClass to `JsonView` so I added check for that. If someone is manually setting the fully namespaced class name or a custom class they should update this check accordingly. In general the app skeleton is meant to be customized as per ones needs.

**Comment by ADmad** [comment]
Turned on serialization for XmlView too. 

Instead of checking `viewClass` how about checking for response type instead?

File: src/Controller/AppController.php [hunk 2]
```diff
@@ -39,4 +40,17 @@ public function initialize()
         parent::initialize();
         $this->loadComponent('Flash');
```
Comment (lorenzo) [inline src/Controller/AppController.php]: Should we add RequestHandler ?

Comment (dereuromark) [inline src/Controller/AppController.php]: Makes sense.

Comment (ADmad) [inline src/Controller/AppController.php]: Yup,  will do.

**Comment by ADmad** [comment]
Loaded `RequestHandler`.

**Comment by markstory** [comment]
This looks good :+1:

**Comment by lorenzo** [comment]
@ADmad I just merged this, but the second after I thought a problem with this. This will override any `_serialize` already set from the controller action :(

**Comment by ADmad** [comment]
@lorenzo Fixing it.

**Pull request merged**